gossip_test.go

```diff
@@ -22,7 +24,7 @@ var _ gossipConnection = &mockGossipConnection{}
 
 func newTestRouter(name string) *Router {
 	peerName, _ := PeerNameFromString(name)
-	router := NewRouter(Config{}, peerName, "nick", nil)
+	router := NewRouter(Config{}, peerName, "nick", nil, log.New(ioutil.Discard, "", 0))
```

Comment:
- There is no leveled logging; all logs are debug logs.
- This PR specifically eliminates package global loggers.

Given both of these things, I'll take that as a tepid approval. Thanks for the commentary.